service/resource/configmap/desired.go

```diff
@@ -47,7 +47,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Maskf(err, "an error occurred creating scrape configs")
 	}
 
-	newPrometheusConfig, err := prometheus.ConfigMerge(*prometheusConfig, scrapeConfigs)
+	newPrometheusConfig, err := prometheus.UpdateConfig(*prometheusConfig, scrapeConfigs)
```

Comment:
`ConfigMerge` became `UpdateConfig`, as it needs to take care to remove old services, and not bork other jobs now.